modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpDispatcher.java

```diff
@@ -135,25 +156,35 @@ public static HttpResource findResource(HTTPServicesRegistry servicesRegistry,
                 Constants.PROTOCOL_PACKAGE_HTTP, Constants.REQUEST);
         BStruct response = ConnectorUtils.createStruct(httpResource.getBalResource(),
                 Constants.PROTOCOL_PACKAGE_HTTP, Constants.RESPONSE);
+
         HttpUtil.setHeaderValueStructType(ConnectorUtils.createStruct(httpResource.getBalResource(),
                 Constants.PROTOCOL_PACKAGE_HTTP, Constants.HEADER_VALUE_STRUCT));
         HttpUtil.populateInboundRequest(request, httpCarbonMessage);
         HttpUtil.populateOutboundResponse(response, HttpUtil.createHttpCarbonMessage(false), httpCarbonMessage);
 
         List<ParamDetail> paramDetails = httpResource.getParamDetails();
-        Map<String, String> resourceArgumentValues =
-                (Map<String, String>) httpCarbonMessage.getProperty(Constants.RESOURCE_ARGS);
-
         BValue[] bValues = new BValue[paramDetails.size()];
         bValues[0] = request;
         bValues[1] = response;
         if (paramDetails.size() <= 2) {
             return bValues;
         }
+
```

Comment:
Can paramDetails.size() < 2 ? According to the above check it seems it is possible. If the size is < 2, the assignment for bValues[0] or bValues[1] may result in IndexOutofBoundsException .

Reply:
Well, it cannot be less that two. It must to be more two or more.